Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Controllers/MixedRealityControllerVisualizer.cs

```diff
@@ -0,0 +1,283 @@
+    /// Once ready, then simply add this script to a GameObject in the scene, for example the Camera / head.
+    /// </example>
+    /// <remarks>
+    /// For Alpha, this visualizer only does basic rendering and management of controllers.  In future versions, it will expose specific transforms on a controller model (if it has them) to be able to attach items to any part of the model
+    /// It will also support animation of controller parts and possibly fading in and our of controllers.
+    /// </remarks>
+    /// <seealso cref="MixedRealityControllerMappingProfile"/>
+    public class MixedRealityControllerVisualizer : InputSystemGlobalListener, IMixedRealitySourcePoseHandler, IMixedRealityInputHandler
+    {
+
+        #region Private Variables
+
+        private static GameObject leftControllerModel;
+        private static IMixedRealityController leftController;
+        private static MixedRealityPose leftControllerOffsetPose = MixedRealityPose.ZeroIdentity;
+        private static GameObject rightControllerModel;
+        private static IMixedRealityController rightController;
+        private static MixedRealityPose rightControllerOffsetPose = MixedRealityPose.ZeroIdentity;
+
+        #endregion Private Variables
+
+        #region IMixedRealitySourcePoseHandler Implementation
+
+        /// <summary>
+        /// The selected controller is moving in the scene, update it's pose in relation to the users movements
+        /// </summary>
+        /// <param name="eventData"></param>
+        void IMixedRealitySourcePoseHandler.OnSourcePoseChanged(SourcePoseEventData eventData)
+        {
+            // Update the respective controller if it has been initialized
+            switch (eventData.Controller.ControllerHandedness)
+            {
+                case Handedness.Left:
+                    if (leftControllerModel != null)
+                    {
+                        leftControllerModel.transform.localPosition = eventData.MixedRealityPose.Position + leftControllerOffsetPose.Position;
+                        leftControllerModel.transform.localRotation = eventData.MixedRealityPose.Rotation * leftControllerOffsetPose.Rotation;
+                    }
+                    break;
+                case Handedness.Right:
+                    if (rightControllerModel != null)
+                    {
+                        rightControllerModel.transform.localPosition = eventData.MixedRealityPose.Position + rightControllerOffsetPose.Position;
+                        rightControllerModel.transform.localRotation = eventData.MixedRealityPose.Rotation * rightControllerOffsetPose.Rotation;
+                    }
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Controller found, create the configured model and position it in the scene
+        /// </summary>
+        /// <param name="eventData"></param>
+        void IMixedRealitySourceStateHandler.OnSourceDetected(SourceStateEventData eventData)
```

Comment:
Have you gotten this to work? I never receive this event for a controller.

The issue seems to be in OpenVRDeviceManager, where the event only fires if a controller hasn't been added to the active controllers list. From my observation, it always gets added first in the `nodeAdded` event, so no event fires.

![image](https://user-images.githubusercontent.com/3580640/43291084-0ad6226a-90e5-11e8-8f2f-c7eac6ff2154.png)

Also, is there a reason why OnSourceDetected/Lost is raised based on the tracking state? It makes more sense to me to be keyed off the nodeAdded/Removed events.

Reply:
That should be an issue against the devices and not the visualization PR.
But in any case, it's against Tracked as that is when the SDK starts sending pose / button updates.  In a non-tracked state, you only get buttons.

Reply:
And yes, works fine for me :D

Reply:
Yup, checked the prefabs and they have the default materials and lighting that come with the controller
![image](https://user-images.githubusercontent.com/1793042/43292912-813675e6-912f-11e8-998c-cb50239b6c40.png)

Reply:
The material has no textures.

![image](https://user-images.githubusercontent.com/3580640/43293069-32346bfc-90ed-11e8-8070-ae1475462a54.png)

Reply:
> But in any case, it's against Tracked as that is when the SDK starts sending pose / button updates. In a non-tracked state, you only get buttons.

Right, but a source is still there. We handle tracked state in the controller implementation, separate from a source existing.

Reply:
Got 'em, merged and done